roles/kernelci-storage/templates/kci-storage.conf

```diff
@@ -0,0 +1,78 @@
+server {
+    listen *;
+    listen [::];
+
+    server_name {{ kci_storage_fqdn }};
+    root /var/www/images/kernel-ci;
+    charset utf-8;
+
+    access_log /var/log/nginx/{{ kci_storage_fqdn }}-access.log combined buffer=16k;
+    error_log /var/log/nginx/{{ kci_storage_fqdn }}-error.log crit;
+
+    if ($host != "{{ kci_storage_fqdn }}") {
+        return 403;
+    }
+
+    # For letsencrypt auto-renew to work.
+    location /.well-known/ {
+        root /usr/share/nginx/html/{{ kci_storage_fqdn }}/;
+    }
+
+    location / {
+        if (-f $document_root/maintenance.html) {
+            return 503;
+        }
+
+{% if ansible_distribution != "CentOS" %}
+        fancyindex on;
+        fancyindex_exact_size off;
+{% endif %}
+    }
+}
+
+{% if role == "production" %}
+server {
+    listen *:443 http2 ssl;
+    listen [::]:443 http2 ssl;
+
+    ssl_certificate /etc/letsencrypt/live/{{ kci_storage_fqdn }}/fullchain.pem;
+    ssl_certificate_key /etc/letsencrypt/live/{{ kci_storage_fqdn }}/privkey.pem;
```

Comment:
There is a variable, called `certname`that needs to be defined, usually is the same as the name of the domain, but you could call the certificate whatever you want.

It should go into a file in the `host_var` directory named after the FQDN.
Look at the template for the nginx configuration of api.kernelci.org.

Reply:
since certname is already used, I wlll use storage_certname

Reply:
Yeah, it is supposed to be used for exactly that purpose.

`certname` is defined in each file in the `host_vars/` directory, since we have `api.kernelci.org` and `staging-api.kernelci.org`.

The storage system should have a `host_vars/` file as well with `certname` defined in it.